### PR TITLE
Fixed bug in opencascade/boundary_lib.cc 

### LINKED ISSUE #862: Failing tests wrt changes in boundary lib
Luca,

can you have a look at those failing tests, please? [1]
```
opencascade/arclength_boundary_02.release 	Failed 	9s 620ms 	Completed (Failed)
opencascade/plane_intersection_01.release 	Failed 	10s 190ms 	Completed (Failed)
opencascade/arclength_boundary_01.release 	Failed 	8s 340ms 	Completed (Failed)
```

[1] http://cdash.kyomu.43-1.org/viewTest.php?onlyfailed&buildid=6147

### PR DESCRIPTION
**Pull request by luca-heltai**
Fixes #862

### CODE REVIEW

File: source/opencascade/boundary_lib.cc [hunk 1]
```diff
@@ -233,13 +233,13 @@ namespace OpenCASCADE
...
     ShapeAnalysis_Curve curve_analysis;
     gp_Pnt proj;
-    Assert(curve_analysis.Project(curve->GetCurve(), point(space_point), tolerance, proj, t, true)
-           < tolerance*length, ExcPointNotOnManifold(space_point));
+#ifdef DEBUG
+    Assert(curve_analysis.Project(curve->GetCurve(), point(space_point), tolerance, proj, t, true) <
+           tolerance*length, ExcPointNotOnManifold(space_point));
 #else
-    (void)space_point;
+    curve_analysis.Project(curve->GetCurve(), point(space_point), tolerance, proj, t, true);
```
Comment (bangerth) [inline source/opencascade/boundary_lib.cc]: Ah, can we avoid this somehow? I.e., just execute the code always and only test its return value in the Assert. Executing instructions with side effects inside Assert statements is bound to be confusing, as you yourself have just found out.

Comment (luca-heltai) [inline source/opencascade/boundary_lib.cc]: We could avoid this, at the price of having a warning at compilation time for release mode, where distance is not used...

**Comment by luca-heltai** [comment]
I've removed the #ifdef and added a bogus operation to silence warnings on Release mode. Le me know if you want me to switch back to what it was before...

File: source/opencascade/boundary_lib.cc (hunk 1 above)
Comment (bangerth) [inline source/opencascade/boundary_lib.cc]: To make the code easier to read, I'd much rather have something like
```
  result = computation();
  Assert (result == ..., Exc...());
#ifndef DEBUG
  (void)result;
#endif
```
At the very least, it is clear that one has to change only one place if the `computation()` call needs to be adjusted later on. The style where you write it twice simply invites a divergence of code.

File: source/opencascade/boundary_lib.cc [hunk 2]
```diff
@@ -233,14 +233,11 @@ namespace OpenCASCADE
...
     ShapeAnalysis_Curve curve_analysis;
     gp_Pnt proj;
-    Assert(curve_analysis.Project(curve->GetCurve(), point(space_point), tolerance, proj, t, true)
-           < tolerance*length, ExcPointNotOnManifold(space_point));
-#else
-    (void)space_point;
-#endif
+    double dist = curve_analysis.Project(curve->GetCurve(), point(space_point), tolerance, proj, t, true);
+    Assert(dist < tolerance*length, ExcPointNotOnManifold(space_point));
+    dist *= 2; // Silence compiler warning in Release mode.
```
Comment (bangerth) [inline source/opencascade/boundary_lib.cc]: Just change this to `(void)dist;` and make the variable `const` -- this is the standard idiom we use everywhere. Then please merge!

**Pull request merged**